### ISSUE TITLE
Snapping on the up/Y axis to help align objects when we create interior only levels

### ISSUE THREAD

**Issue by MonkeyFrogStudio** [Dark-Basic-Software-Limited/GameGuruRepo#1912]
Some of us want to make interior only settings, like with the original DOOM game. As such, we need a few simple settings to do this. For example, we need a SIMPLE way to totally remove the sky, such a as a simple check box or button to TOTALLY remove ALL sky elements, setting the sky to pitch black. Currently, using the dropdown box and selecting NONE for the sky still leaves you with a sun and a dozen other settings (ambient, etc.) that you have to set. If you want to create an interior only level, then it would be great to have only ONE button or check box to click or check to TOTALLY remove the sky as it is not at all needed. 

Secondly, as has been asked for several times before by several other users, have an option for a true EMPTY level with not terrain at all, not even the flat grid plain. For an indoor-only level, we don't need it. When building an interior only level, we want to have the ability to build our levels both upward and downward as we please without having to carve a hole in the terrain to do so.

To expedite this, it would be great to just have an interior-only selection added to the "bioms". By selecting this, the end-user is taken to the editor and presented with a blank screen in the viewport - no sky, no flat grid terrain (only a grid like in a 3D program indicating where the world axis 0,0,0) - so they can just begin creating their indoor level. 

Not all of us want to create outdoor levels all of the time for each of our levels. Some of us want to create DOOM and Half-life like games. Some of us want to create games on spaceships, etc. 

Thanks.

**Comment by UltraVox001** [comment]
Yes, an "instant" setting (Special Biome) for indoor maps would be appreciated.
No sky, no sun or tiled ground. No light of any kind.
Absolute emptiness, like when you want to use EBE.

**Comment by ArgentArts** [comment]
... just don't force us to use EBE if you give us this truly empty setting. ;) EBE as an option. ;)

**Comment by Necrym59** [comment]
Try using GZDoom_Builder to make an indoor level you can change and use your own textures then use the export to object it will give you the obj file and all the texures used ready for flawless importing into MAX then just add your decorations, doors, lights etc.

**Comment by ArgentArts** [comment]
No thank you. Firstly, OBJs are very limited ... only one UV map per OBJ. Secondly, this has nothing to do with my request, which has to do with how one works in the actual MAX editor itself while creating indoor levels there. Thirdly, I create my own models and materials and can handle that quite well on my own. I don't need tool like GZDoom_Builder to, frankly, get in my way. It would actually hinder me in that respect. But I do appreciate the suggestion. I truly do.

**Comment by Kitakazi** [comment]
Yes, please there is no need for excess things active when working on a purely interior level. Currently unless you have all the lighting to 0,0,0 the sun will still light the scene in whichever direction it is facing. Even indoors...

Additionally having the terrain actually be blank or invisible is needed for skyboxes that have detail on the underside. There is default scripts in classic HideTerrain() and HideWater() and completely hide terrain. Hide and Show Terrain/water should be default scripts with max...

**Comment by ArgentArts** [comment]
> Hide and Show Terrain/water should be default scripts with max...

Actually, we shouldn't need scripts for this. There should just be a setting in the editor to start without them if you know your current level is not going to need them. We have all these things at the beginning asking us how we'd like to start out our level - rain forest, etc. Why not an interior level "biom", too? No sun, no terrain, no water ... just ready to lay down some interiors and get ready to rumble in old-school FPS goodness! ;)

**Comment by LeeBamberTGC** [comment]
@MonkeyFrogStudio Added hide terrain to view options :)

**Comment by MonkeyFrogStudio** [comment]
@LeeBamberTGC We can no hide the terrain (thank you), but the camera still can't go below where the terrain was. So, if we were to make 'sub levels', we couldn't get to them in the editor.

**Comment by MonkeyFrogStudio** [comment]
@LeeBamberTGC What we really need is No Terrain at all (not just an invisible one) along with Snapping on the up/Y axis to help align objects when we create interior only levels.

**Comment by LeeBamberTGC** [comment]
@MonkeyFrogStudio The other issue seems to be dominant in respect to the empty empty biome so we can use that for the remains of this: https://github.com/TheGameCreators/GameGuruRepo/issues/2113

I have changed the title to keep the suggestion of the Y snap which may well be resolve along the way (when we move the grid properties to the top bar).

**Comment by MonkeyFrogStudio** [comment]
I'm closing this because we now have empty levels and snapping on the Y axis.
